packages/shared/src/lib/element.ts

```diff
@@ -62,3 +63,9 @@ export const isCompanionActivated = (): boolean =>
   !!document
     .querySelector('daily-companion-app')
     ?.shadowRoot?.querySelector?.('#daily-companion-wrapper')?.firstChild;
+
+export interface OptionMenuItem {
+  icon: ReactElement;
+  text: string;
+  action: () => unknown;
+}
```

Comment:
I think we have an existing OptionItem menu interface in the `PortalMenu.tsx`.

Reply:
The interface in the `PortalMenu.tsx` returns `Promise<void>` which is different to the above. Possibly the action should return void instead of unknown. I can make the change however, will need to update 5 or 6 functions to match that return type so just want to make sure thats the right thing to do?

Reply:
Yes, I agree. I think it would be better to update the type to `() => unknown  | Promise<unknown>`.

Or even better, make it generic to accept what type it would return with a default of unknown. WDYT?